### ISSUE TITLE
0.12.0.3 release candidate 

### ISSUE THREAD

**Issue by cartazio** [haskell/vector#242]
please check https://hackage.haskell.org/package/vector-0.12.0.3/candidate

@hvr @RyanGlScott  :)

**Comment by RyanGlScott** [comment]
It builds on GHC 8.8, so it's good from my perspective. You might consider whether you want to define `Monad.fail = Fail.fail` instead of `Monad.fail _ = empty` (see https://github.com/haskell/vector/issues/241#issuecomment-484598549), but that's by no means a blocker.

**Comment by cartazio** [comment]
@RyanGlScott  i updated it (after a momentary confusion), please confirm the fixedup one is good to go per your warning suggestion etc :)

**Comment by RyanGlScott** [comment]
Commit 998b18485cf386ef0300ec3ee9b656a3cf4f4a4b [doesn't compile](https://travis-ci.org/haskell/vector/builds/526133761):

```
[13 of 21] Compiling Data.Vector      ( Data/Vector.hs, interpreted )

Data/Vector.hs:350:12: error:
    • Couldn't match expected type ‘Vector a’
                  with actual type ‘String -> m0 a0’
    • Probable cause: ‘Fail.fail’ is applied to too few arguments
      In the expression: Fail.fail
      In an equation for ‘Fail.fail’: Fail.fail _ = Fail.fail
      In the instance declaration for ‘Fail.MonadFail Vector’
    • Relevant bindings include
        fail :: String -> Vector a (bound at Data/Vector.hs:350:3)
    |
350 |   fail _ = Fail.fail -- == empty vector
    |            ^^^^^^^^^
```

I think you meant `fail = Fail.fail`, not `fail _ = Fail.fail`.

**Comment by cartazio** [comment]
@RyanGlScott thats the commit before that one i actually have,.

i tried to force push the fix commit to github. let me try again :)

**Comment by cartazio** [comment]
@RyanGlScott i deleted v12 and repushed the version on hackage candidate. please test the hackage candidate one :)

**Comment by RyanGlScott** [comment]
Thanks! Everything looks ready to go now.

**Comment by cartazio** [comment]
done!

**Comment by cartazio** [comment]
@RyanGlScott Did i forget to relax the bounds before cutting the release?

**Comment by RyanGlScott** [comment]
That appears to be the case, yes. Revision time?

**Comment by cartazio** [comment]
YUPPP

**Comment by cartazio** [comment]
did it :)

**Comment by RyanGlScott** [comment]
Thanks! As one last bit of cleanup, I suppose the new commits to the `v0.12` branch (e.g., the `base` version bump) should also be cherry-picked to `master`?

**Comment by cartazio** [comment]
i already had a commit setup, so done!

**Comment by cartazio** [comment]
i actually need to forward port 1-2 patches in the 0.12.0 branch, namely /
most especially the Addr vs pointer cleanup in storable. i dont think i
applied that to master yet
